.github/workflows/update-rules.yml

```diff
@@ -0,0 +1,57 @@
+name: Update AppSec Rules
+on:
+  workflow_dispatch: # manually
+  schedule:
+    - cron: "0 6 * * 1" # every Monday at 06:00 UTC
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    outputs:
+      mutation_happened: ${{ steps.detect.outputs.mutation_happened }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+      - name: Upgrade AppSec Rules
+        run: ./_tools/rules-updater/update.sh
+      - name: Detect Udpated Coded
+        id: detect
+        run: |-
+          git add .
+          git diff --staged --patch --exit-code > ${{ runner.temp }}/repo.patch || echo "mutation_happened=true" >> $GITHUB_OUTPUT
+      - name: Upload Patch
+        if: steps.detect.outputs.mutation_happened
+        uses: actions/upload-artifact@v3
+        with:
+          name: repo.patch
+          path: ${{ runner.temp }}/repo.patch
+
+  create-pr:
+    needs: update
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
```

Comment:
This is the reason why this is separated from actually creating the patch :)

Reply:
Very cool GH feature - I wasn't aware of it